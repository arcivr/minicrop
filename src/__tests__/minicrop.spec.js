const Minicrop = require('../minicrop.js')

describe("Minicrop", () => {
  var root
  var cropper

  beforeEach(() => {
    document.body.innerHTML = "<div class='minicrop' data-image='image'></div>"
    root = document.getElementsByClassName('minicrop')[0]
    cropper = new Minicrop.default(root)
  })

  test("result", () => {
    expect(cropper.element)
      .toMatchSnapshot()
  })

  describe("crop", () => {
    test("response", () => {
      expect(cropper.crop())
        .toMatchSnapshot()
    })
  })
})
