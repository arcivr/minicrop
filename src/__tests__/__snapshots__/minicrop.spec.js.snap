// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Minicrop crop response 1`] = `
Object {
  "height": 0,
  "ratio": 1,
  "rotated": false,
  "width": 0,
  "x": -40,
  "y": -40,
}
`;

exports[`Minicrop result 1`] = `
<div
  class="minicrop"
  data-image="image"
>
  
    
  <img
    class="image"
    height="100%"
    src="image"
    style="top: 0px; left: 0px;"
  />
  
    
  <div
    class="crop"
  >
    
      
    <!-- &lt;img src="image" class="image-preview" height="100%" /&gt; -->
    
    
  </div>
  
  
</div>
`;
